File: vacation_backend/Application/DTOs/Employee/EmployeeListDto.cs
namespace vacation_backend.Application.DTOs.Employee
{
    public class EmployeeListDto
    {
        public int Id { get; set; }

        // Información básica
        public string FullName { get; set; } = null!;

        // Información laboral
        public string DepartmentName { get; set; } = null!;
        public string RoleName { get; set; } = null!;
        public string Status { get; set; } = null!;

        // Días de vacaciones
        public int AvailableDays { get; set; }
        public int UsedDays { get; set; }
        public int RemainingDays { get; set; }

        // Días extra de beneficio
        public int RemainingExtraBenefitDays { get; set; }

    }
}

File: vacation-frontend/src/models/employee/EmployeeListDto.ts
import type { Item } from '../ItemModel'

export interface EmployeeListDto {
  id: number
  firstName: string
  lastName: string
  email?: string | null

  department: Item
  role: Item
  status: Item

  availableDays: number
  usedDays: number
  remainingDays: number
  remainingExtraBenefitDays: number
}